internals/cli/cli.go

```diff
@@ -165,37 +146,50 @@ func fixupArg(optName string) string {
 	return optName
 }
 
-type clientSetter interface {
-	setClient(*client.Client)
-}
-
 type clientMixin struct {
 	client *client.Client
 }
 
+type clientSetter interface {
+	setClient(*client.Client)
+}
+
 func (ch *clientMixin) setClient(cli *client.Client) {
 	ch.client = cli
 }
 
+type parserSetter interface {
+	setParser(*flags.Parser)
+}
+
+type options struct {
+	Version func() `long:"version"`
+}
+
 // Parser creates and populates a fresh parser.
 // Since commands have local state a fresh parser is required to isolate tests
 // from each other.
 func Parser(cli *client.Client) *flags.Parser {
-	optionsData.Version = func() {
-		printVersions(cli)
-		panic(&exitStatus{0})
+	// Implement --version by default on every command
```

Comment:
Shouldn't `--version` only work when used on the executable without specifying a command?